src/coreclr/vm/gcheaputilities.h

```diff
@@ -40,20 +40,29 @@ typedef struct _ee_alloc_context
 
     inline void ComputeSamplingLimit(CLRRandom* pRandomizer)
     {
-        // TODO: maybe it is easier to assume that the caller of this function will check if sampling is on/off
-        // If sampling is off this is just setting alloc_sampling = alloc_limit
+        // the caller of this function does not have to check if sampling is on/off
+        // If sampling is off, this is just setting alloc_sampling = alloc_limit
         // If sampling is on then we'd do some pseudo-random number generation to decide what is
         // the next sampled byte in the gc_alloc_context, if any.
+        if (EventPipeEventEnabledAllocationSampled())
```

Comment:
```suggestion
        if (IsEventEnabledAllocationSampled())
```
It isn't only EventPipe that can turn this on, ETW or Lttng should be capable of doing it too.

Reply:
ok : I'm replacing by 
```
ETW_TRACING_CATEGORY_ENABLED(MICROSOFT_WINDOWS_DOTNETRUNTIME_PROVIDER_DOTNET_Context,
                                     TRACE_LEVEL_INFORMATION,
                                     CLR_ALLOCATIONSAMPLING_KEYWORD)
```